Sources/SourceKit/sourcekitd/SwiftLanguageServer.swift

```diff
@@ -37,6 +37,27 @@ fileprivate extension Range {
   }
 }
 
+/// Explicitly blacklisted `DocumentURI.pseudoPath` prefixes. No diagnostics will be emitted
+/// for these documents.
+fileprivate let documentPathPrefixBlacklist = [
+  "git",
+  "hg",
+]
+
+/// Returns true if the document's URL is considered blacklisted. Some editors  (like Visual
+/// Studio Code) use non-file URLs to manage source control diff bases for the active document,
+/// which can lead to duplicate diagnostics in the Problems view. As a workaround we explicitly
+/// blacklist those URIs and don't emit diagnostics for them.
+fileprivate func isDocumentBlacklisted(_ document: DocumentURI) -> Bool {
+  let path = document.pseudoPath
```

Comment:
Instead of using string prefixes, how about we expose `scheme: String?` on DocumentURI from the underlying URL? Then we can consistently talk about this as being a list of excluded URI schemes.

Terminology-wise, would it make sense to  say `enableDiagnostics(for:)`, and `excludedDocumentURISchemes`?

Reply:
Thought about it, makes sense, done